bootstrap/postgres/configs.go

```diff
@@ -451,59 +460,86 @@ func (cr configRepository) RemoveChannel(ctx context.Context, id string) error {
 	return nil
 }
 
+func (cr configRepository) ConnectThing(ctx context.Context, channelID, thingID string) error {
+	q := `UPDATE configs SET state = $1 WHERE EXISTS (
+	SELECT 1 FROM connections WHERE config_id = $2 AND channel_id = $3)`
+	result, err := cr.db.ExecContext(ctx, q, bootstrap.Active, thingID, channelID)
+	if err != nil {
+		return errors.Wrap(errConnectThing, err)
+	}
+	if rows, _ := result.RowsAffected(); rows == 0 {
+		return repoerr.ErrNotFound
+	}
+	return nil
+}
+
 func (cr configRepository) DisconnectThing(ctx context.Context, channelID, thingID string) error {
 	q := `UPDATE configs SET state = $1 WHERE EXISTS (
 		SELECT 1 FROM connections WHERE config_id = $2 AND channel_id = $3)`
-	if _, err := cr.db.ExecContext(ctx, q, bootstrap.Inactive, thingID, channelID); err != nil {
+	result, err := cr.db.ExecContext(ctx, q, bootstrap.Inactive, thingID, channelID)
+	if err != nil {
 		return errors.Wrap(errDisconnectThing, err)
 	}
+	if rows, _ := result.RowsAffected(); rows == 0 {
+		return repoerr.ErrNotFound
+	}
```

Comment:
Please explain why we need this error.

Reply:
The error was initially added to inform the caller that the specified `thingID` and `channelID` pair do not exist in Magistrala-platform. However, we have previously agreed that when disconnecting, we don't need to care about the existence of this `thingID` and `channelID`  pair. If they don't exist, it means they are already disconnected. Soooo I'll fix this to return something like a `HTTP status 204` to indicate successful disconnection, whether or not the connection existed.